libs/components/core/src/lib/modules/stacking-context/stacking-context.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Inject, Injectable } from '@angular/core';
+
+import { BehaviorSubject, Observable, isObservable } from 'rxjs';
+
+import {
+  SkyStackingContext,
+  SkyStackingContextInterface,
+} from './stacking-context';
+
+@Injectable()
+export class SkyStackingContextService {
```

Comment:
A service might be an unnecessary layer for this feature. You could simply have a token named `SKY_STACKING_CONTEXT` that uses the `SkyStackingContext` interface. For example:
```
export const SKY_STACKING_CONTEXT = new InjectionToken<SkyStackingContext>(
  'SkyStackingContext'
);
```
The token would be provided in the flyout, overlays, etc:
```
providers: [{
  provide: SKY_STACKING_CONTEXT,
  useValue: {
    zIndex: this.zIndexObs
  }
}]
```
The consuming component would then reference the context like this:
```
constructor(
  @Optional() @Inject(SKY_STACKING_CONTEXT) stackingContext: SkyStackingContext
) {
  stackingContext?.zIndex.pipe().subscribe(...)
}
```

Reply:
Take a look at Angular's `IMAGE_LOADER` token for some ideas: https://github.com/angular/angular/blob/74258f3652eff268c7599c0f71baf929ed5b244b/packages/common/src/directives/ng_optimized_image/image_loaders/image_loader.ts#L106

Reply:
Switched to using the injection token instead of the service.